src/module_PI/Raspberry_PI/main/EmailNotifier.java

```diff
@@ -0,0 +1,239 @@
+package module_PI.Raspberry_PI.main;
+
+import java.io.*;
+import java.util.*;
+
+import javax.activation.*;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+public class EmailNotifier {
+
+	private String gmailUsername;
+	private String gmailPassword;
+
+	private String recipientEmailAddress;
+	private String emailSubject = "Calico Home Security";
+	private String emailMessage;
+	private String attachmentFilePath;
+
+	private Properties mailSessionProperties;
+	private Session javaMailSession;
+	private Message javaMailMessage;
+	private BodyPart messageBodyPart;
+	private Multipart multipart;
+	private DataSource attachmentDataSource;
+
+	public EmailNotifier(String recipientEmailAddress, String message) {
+		this(recipientEmailAddress, message, null);
+	}
+
+	public EmailNotifier(String recipientEmailAddress, String message, String pathToAttachment) {
+		this.loadGmailCredentials();
+		this.setRecipientEmailAddress(recipientEmailAddress);
+		this.emailMessage = message;
+		this.setAttachmentFilePath(pathToAttachment);
+		buildPropertiesAndCreateSession();
+	}
+
+	private void loadGmailCredentials() {
+		try {
+			Properties gmailProperties = new Properties();
+			FileInputStream in = new FileInputStream("gmailCredentials");
```

Comment:
Since you went with the command line arguments you didn't need to interact with a file at all.

You should just create a `private Properties gmailProperties = new Properties();` in main or here when you are parsing the args. 

Then you would do something like 
```java
gmailProperties.put("gmailUsername", args[0]);
gmailProperties.put("gmailPassword", args[1]);
```
to set the values as you are parsing through the args.

When you needed to get the values out you would use something like `gmailProperties.getProperty("gmailUsername");`.

Using the above you never have to interact with a file at all. See https://www.mkyong.com/java/java-properties-file-examples/ for some examples of working with Properties.

Ultimately, this comes down to this: since you are throwing an exception in main for not passing in command line arguments properly having a file that you later read from is just extra work. If you didn't require parameters if you already have a gmailCredentials file _then_ it would be more reasonable to still be reading from a file like this.